src/engine/sparqlExpressions/StdevExpression.cpp

```diff
@@ -4,71 +4,78 @@
 
 #include "engine/sparqlExpressions/StdevExpression.h"
 
-namespace sparqlExpression {
+#include "engine/sparqlExpressions/SparqlExpressionTypes.h"
 
-namespace detail {
+namespace sparqlExpression::detail {
 
 // _____________________________________________________________________________
 ExpressionResult DeviationExpression::evaluate(
     EvaluationContext* context) const {
-  auto impl = [context](SingleExpressionResult auto&& el) -> ExpressionResult {
+  // Helper: Extracts a double or int (as double) from a variant
+  auto numValVisitor = []<typename T>(T& value) -> std::optional<double> {
+    if constexpr (ad_utility::isSimilar<T, double> ||
+                  ad_utility::isSimilar<T, int64_t>) {
+      return static_cast<double>(value);
+    } else {
+      return std::nullopt;
+    }
+  };
+
+  // Helper to replace child expression results with their squared deviation
+  auto devImpl = [context, numValVisitor](
+                     bool& undef,
+                     VectorWithMemoryLimit<IdOrLiteralOrIri>& exprResult,
+                     auto generator) {
+    double sum = 0.0;
+    // Intermediate storage of the results returned from the child
+    // expression
+    VectorWithMemoryLimit<double> childResults{context->_allocator};
+
+    // Collect values as doubles
+    for (auto& inp : generator) {
+      const auto& n = detail::NumericValueGetter{}(std::move(inp), context);
+      auto v = std::visit(numValVisitor, n);
+      if (v.has_value()) {
+        childResults.push_back(v.value());
+        sum += v.value();
+      } else {
+        // There is a non-numeric value in the input. Therefore the entire
+        // result will be undef.
+        undef = true;
+        return;
+      }
+      context->cancellationHandle_->throwIfCancelled();
+    }
+
+    // Calculate squared deviation and save for result
+    double avg = sum / static_cast<double>(context->size());
+    for (size_t i = 0; i < childResults.size(); i++) {
+      exprResult.at(i) = IdOrLiteralOrIri{
+          ValueId::makeFromDouble(std::pow(childResults.at(i) - avg, 2))};
+    }
+  };
+
+  // Visitor for child expression result
+  auto impl = [context,
+               devImpl](SingleExpressionResult auto&& el) -> ExpressionResult {
     // Prepare space for result
     VectorWithMemoryLimit<IdOrLiteralOrIri> exprResult{context->_allocator};
     std::fill_n(std::back_inserter(exprResult), context->size(),
                 IdOrLiteralOrIri{Id::makeUndefined()});
```

Comment:
This seems like a waste of time: Either you write a double to all entries here, and then you don't care about the initialization, or you return a single UNDEF in a special path anyway. 
So you can simply do `resize(context->size)` here (or pass the size directly to the constructor of the `VectorWithMemoryLimit`.